src/inmanta/agent/resourcepool.py

```diff
@@ -337,11 +346,17 @@ async def cleanup_inactive_pool_members_task(self) -> None:
         We split up `cleanup_inactive_pool_members` and `cleanup_inactive_pool_members_task` in order to be able to call the
         cleanup method in the test without being blocked in a loop.
         """
-        while self.running:
-            sleep_interval = await self.cleanup_inactive_pool_members()
-            if self.running:
-                LOGGER.log(LOG_LEVEL_TRACE, "Manager will clean in %.2f seconds", sleep_interval)
-                await asyncio.sleep(sleep_interval)
+        try:
+            while self.running:
+                sleep_interval = await self.cleanup_inactive_pool_members()
```

Comment:
Don't we also want to wrap this method into a try-catch block? Otherwise we might drop out of the while loop when it raises an exception.